### PR TITLE
TST: fix DecimalArray._reduce kludges

### PR DESCRIPTION
**Pull request by jbrockmendel**

### CODE REVIEW

File: pandas/core/groupby/ops.py [hunk 1]
```diff
@@ -663,6 +663,14 @@ def _aggregate_series_pure_python(self, obj: Series, func):
             counts[label] = group.shape[0]
             result[label] = res
 
+            if isinstance(res, np.bool_):
+                if is_extension_array_dtype(group.dtype):
+                    # FIXME: kludge for DecimalArray on numpy<1.15 returning
+                    #  np.bool_(False) when we have all-NA DecimalArray
+                    if group.isna().all():
+                        res = group.dtype.na_value
+                        result[label] = res
```
Comment (jorisvandenbossche) [inline pandas/core/groupby/ops.py]: It's not clear to why this is the correct fix. You are changing the result of the function, without necessarily knowing what the function is?

File: pandas/tests/extension/decimal/array.py [hunk 2]
```diff
@@ -71,7 +71,15 @@ def dtype(self):
 
     @classmethod
     def _from_sequence(cls, scalars, dtype=None, copy=False):
-        return cls(scalars)
+        try:
+            return cls(scalars)
+        except TypeError:
+            # handling int 0 is necessary for groupby tests
```
Comment (jorisvandenbossche) [inline pandas/tests/extension/decimal/array.py]: Can you explain why this is necessary? What is a code example running into this?

**Review by jorisvandenbossche — CHANGES_REQUESTED** [review]

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: yah, i want to take another shot at fixing this behavior inside `DecimalArray.__array_whatever__`

File: pandas/tests/extension/decimal/array.py (hunk 2 above)
Comment (jbrockmendel) [reply pandas/tests/extension/decimal/array.py]: this is made unnecessary by the fix i just pushed, so has been reverted

File: pandas/core/groupby/ops.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/groupby/ops.py]: tentatively looks like a fix in `DecimalArray._reduce` gets the job done, so the change here has been reverted.  We'll see if the CI agrees.

**Review by jreback — APPROVED** [review]
lgtm. @jorisvandenbossche

**Comment by jbrockmendel** [comment]
@jorisvandenbossche gentle ping

**Review by TomAugspurger — APPROVED** [review]

**Pull request merged**